data.go

```diff
@@ -283,6 +287,109 @@ func (d *Data) IsObject() bool {
 	return d.NativeTypeNum == C.DPI_NATIVE_TYPE_OBJECT
 }
 
+// NewData returns Data for input parameters on Object/ObjectCollection.
+func (c *conn) NewData(baseType interface{}, sliceLen, bufSize int) ([]*Data, error) {
+	if c == nil || c.dpiConn == nil {
+		return nil, errors.New("connection is nil")
+	}
+
+	var natType C.dpiNativeTypeNum
+	var oraType C.dpiOracleTypeNum
+	var objectType *C.dpiObjectType
+	var obj ObjectType
+
+	switch v := baseType.(type) {
+	case Lob, []Lob:
+		natType = C.DPI_NATIVE_TYPE_LOB
```

Comment:
case []Lob:	
                isSlice = True
                fallthrough
case Lob: 
		natType = C.DPI_NATIVE_TYPE_LOB
		var isClob bool
		switch v := v.(type) {
		case Lob:
			isClob = v.IsClob
		case []Lob:
			isClob = len(v) > 0 && v[0].IsClob
		}		
		if isClob {
			oraType = C.DPI_ORACLE_TYPE_CLOB
		} else {
			oraType = C.DPI_ORACLE_TYPE_BLOB
		}

Reply:
Is it possible to "fallthrough"  in a type switch?

Reply:
You're right, type switch is not a regular switch, forgot it.